blog/2023-06-12-bringing-dlc-to-lightning-part-1.md

```diff
@@ -0,0 +1,82 @@
+
+The following diagram illustrates the transaction structure of a lightning channel.
+
+![](/2023-06-12-bringing-dlc-to-lightning-part-1/lightning_channel.png)
+
+## Adaptor Signatures
+
+Adaptor signatures are a cryptographic technique that allows for the construction of interactive protocols where two parties can collaboratively create a signature on a specific message without revealing their private keys to each other. This concept has various applications, including in the context of Discreet Log Contracts (DLCs).
+
+In a traditional digital signature scheme, a signer generates a signature using their private key and shares the resulting signature with a verifier. The verifier can then use the signer's public key to verify the authenticity and integrity of the message. However, in the case of adaptor signatures, the process is more nuanced.
+
+In an adaptor signature scheme, the signer generates a partial signature that is dependent on some secret value or nonce. This partial signature alone is not valid and cannot be verified by the verifier. The signer then provides this partial signature to the verifier, who completes the signature by adding their own contribution to it. The resulting adapted signature is a valid signature on the message. However, the verifier does not gain any knowledge of the signer's secret value or nonce during this process.
+
+In the context of DLCs, adaptor signatures allow for the creation of smart contracts where the settlement of funds is contingent on an outcome determined by an external oracle. The parties can collaboratively construct a signature on the outcome without revealing their private keys to each other. This ensures security and privacy while enabling the execution of self-enforcing contracts.
+
+![](/2023-06-12-bringing-dlc-to-lightning-part-1/adaptor_signature.png)
+
+Overall, adaptor signatures provide a powerful tool for secure and collaborative signature generation, enabling the creation of advanced cryptographic protocols with enhanced privacy and flexibility. The following figure illustrates the different operations involved in encrypting, verifying, and decrypting adaptor signatures, as well as recovering an encryption secret.
+
+## Recap on DLCs
+
+Discreet Log Contracts (DLCs) are a type of smart contract protocol that operates on the Bitcoin blockchain, allowing for the execution and settlement of contracts based on external events.
+
+While resembling Lightning channels at first glance, DLCs differ in their transaction structure. Instead of utilizing commitment transactions to spend from the funding output, DLCs employ Contract Execution Transactions (CETs), with each CET corresponding to a potential contract payout outcome.
+
+To construct these CETs, the involved parties encrypt their signatures using adaptor signatures, incorporating the pre-image of the signature that the oracle would generate for the specific event outcome. The parties exchange and verify these encrypted signatures.
```

Comment:
This sounds wrong.

I would say that since the CETs spend from the shared DLC output, they require a signature from each party to be minable. Each party can always provide their own, but they need the counterparty's too. During the DLC setup, for each CET, they also get an adaptor signature corresponding to the other party's signing key, encrypted under the [anticipated signature (or signature point)](https://github.com/coblox/academy/blob/master/2021-05-19_dlcs/src/lib.rs#L274-L288) that corresponds to a particular potential event outcome

Reply:
I've rewritten the whole paragraph, please have a look again. Not sure if I got it 100% right :)